libs/wingc/src/type_check.rs

```diff
@@ -3944,6 +3964,29 @@ impl<'a> TypeChecker<'a> {
 							)
 						}
 					}
+					Type::Struct(ref s) => {
+						if property.name == "fromJson" {
+							// we need to validate that only structs with all valid json fields can have a fromJson method
+							for (name, field) in s.fields(true) {
+								if !field.has_json_representation() {
+									self.spanned_error_with_var(
+										property,
+										format!(
+											"Struct \"{}\" contains a non-json compatible field \"{}\" and cannot call fromJson method",
+											type_, name
+										),
+									);
+									return (self.make_error_variable_info(), Phase::Independent);
+								}
+							}
+						}
```

Comment:
Well initially I added it because I swore it was in the spec, but Im not seeing it at the moment 😁 (maybe its in some thread). Though Im torn because I dislike adding the special case checks, but I think there is a pretty big value add in giving the user a heads up in compilation time versus allowing them to march forward to a runtime error. Especially in the case of when I add `tryFromJson` the user may get pretty confused as to why they always get undefined when trying to parse.

Reply:
Ok let's keep this approach, the verbose error is worth it I think.